devito/ops/node_factory.py

```diff
@@ -0,0 +1,214 @@
+from sympy import Eq, Add, Mul, Pow, Integer, Float, Rational
+from devito import Dimension
+from devito.symbolics import Macro
+from devito.types import Symbol, Indexed, Array
+
+from devito.ops.utils import namespace
+
+
+class OpsNodeFactory():
+    """
+    Generates ops nodes for building an OPS expression.
+
+    Examples
+    --------
+    >>> a = OpsNodeFactory()
+    >>> b = a.new_symbol('symbol_name')
+    """
+
+    def __init__(self):
+        self.ops_grids = {}
+
+    def new_symbol(self, name):
+        """
+        Creates a new sympy :class:`Symbol` object with the given name.
+        Its used to define/use variables.
+
+        Parameters
+        ----------
+        name : str
+            Name of the symbol to be created.
+
+        Returns
+        -------
+        :class:`Symbol` 
+            Object :class:`Symbol` with the given name.
+        """
+        return Symbol(name=name)
+
+    def new_int_node(self, number):
+        """
+        Creates a new sympy :class:`Integer` object.
+
+        Parameters
+        ----------
+        number : int
+            Number to initialize an Integer node.
+
+        Returns
+        -------
+        :class:`Integer`
+            Object of :class:`Integer`.
+        """
+        return Integer(number)
+
+    def new_float_node(self, number):
+        """
+        Creates a new sympy :class:`float` object.
+
+        Parameters
+        ----------
+        number : float
+            Number to initialize an Float node.
+
+        Returns
+        -------
+        :class:`Float`
+            Object of :class:`Float`.
+        """
+        return Float(number)
+
+    def new_rational_node(self, num, den):
+        """
+        Creates a new sympy :class:`Rational` object to represent 
+        a mathematical rational.
```

Comment:
I am not sure if the expression _mathematical rational_  can stand alone... Mathematical rational what?
Probably I am missing something but I do not quite get it.

Reply:
I see your point. what about: "mathematical rational operation" or just "rational operation" ?

Reply:
Whatever, its up to you.